### PR TITLE
[FE-41] Enable adding points for teacher

### PR DESCRIPTION
**Pull request by AnCichocka**
I have some problems with formik validation connected with maxPoints and displaying form errors in general. Overall I would like to merge it, because adding points works and returns query errors correctly (if they occurs). I would like to create separate ticket for correcting this form - for now I don't have any better idea, maybe I will resign from formik with this particular form eventually....
- correct form so it uses only options from current edition
- adding points for student by teacher works with refetching works!

https://github.com/user-attachments/assets/dcd5e477-6807-46d9-924d-bbbc0e10d442

additionaly:
- some infrastructure clean up, big hooks coming
- improved goups list / group screen (it helped me with testing)

image

image

Sorry for a little big PR....

### CODE REVIEW

File: frontend/src/hooks/Groups/useGroupsData.tsx [hunk 1]
```diff
@@ -1,31 +1,25 @@
...
 
-  const { data, loading, error } = useGroupsByEditionQuery({
+  const { data, loading, error } = useGroupsListByEditionQuery({
     variables: {
       editionId: selectedEdition?.editionId as string,
     },
     skip: !selectedEdition,
   });
 
+  // TODO there must be a better way to get teacherId
```
Comment (AnCichocka) [inline frontend/src/hooks/Groups/useGroupsData.tsx:15]: @Szyntos ?

Comment (Szyntos) [reply frontend/src/hooks/Groups/useGroupsData.tsx:15]:
```js
const groups: Group[] | undefined = data?.editionByPk?.groups.map((group) => {
    return {
      name: group.groupName,
      id: group.groupsId,
      teacherId: group?.userGroups[0]?.user.filter((user) => {
        user.role != "student"
      }
    };
  });
  ```

File: frontend/src/screens/StudentProfile/hooks/useCategoriesCardData.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    refetch: categoriesRefetch,
+  } = useCategoriesPointsQuery({
+    variables: {
+      editionId: parseInt(editionId),
+      studentId: parseInt(studentId),
+    },
+  });
+
+  // why so many undefines here
+  const categories: PointsBarProps[] =
```
Comment (AnCichocka) [inline frontend/src/screens/StudentProfile/hooks/useCategoriesCardData.tsx:23]: probably to correct after merge main

File: frontend/src/hooks/Groups/useGroupsData.tsx (hunk 1 above)
Comment (Szyntos) [reply frontend/src/hooks/Groups/useGroupsData.tsx:15]: i can add a field in Groups with teacherId, but i don't think it's that necessary

File: frontend/src/components/form/PointsForm/PointsForm.tsx [hunk 3]
```diff
@@ -67,52 +87,48 @@ export const PointsForm = ({ handleAddPoints: handleAdd }: PointFormProps) => {
...
+          name="subcategoryId"
+          optionItems={subcategories?.map((subcategory) => ({
+            value: subcategory.subcategoryId,
+            title: subcategory.subcategoryName,
+          }))}
+          label="subcategory"
+        />
+        <NumberInput
+          handleChange={formik.handleChange}
+          handleBlur={formik.handleBlur}
```
Comment (Szyntos) [inline frontend/src/components/form/PointsForm/PointsForm.tsx:122]: [Let me be clear](https://www.youtube.com/watch?v=9pHlguBNxQk)

File: frontend/src/hooks/Groups/useGroupsData.tsx (hunk 1 above)
Comment (AnCichocka) [reply frontend/src/hooks/Groups/useGroupsData.tsx:15]: ok, I have other problem why I didn't wanted to add another check for students, but now I see I was kinda stupid, nevermind, all good :<

File: frontend/src/hooks/Groups/useStudentsSearchData.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { useStudentsInGroupQuery } from "../../graphql/studentsInGroups.graphql.types";
+import { ShortStudent } from "../../utils/types";
```
Comment (Szyntos) [inline frontend/src/hooks/Groups/useStudentsSearchData.tsx:2]: THAT'S MEEEEE

MOM, I'M FAMOUS

File: frontend/src/hooks/StudentProfile/useCategoriesCardData.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    error: categoriesError,
+    refetch: categoriesRefetch,
+  } = useCategoriesPointsQuery({
+    variables: {
+      editionId: parseInt(editionId),
+      studentId: parseInt(studentId),
+    },
+  });
+
+  // why so many undefines here
```
Comment (Szyntos) [inline frontend/src/hooks/StudentProfile/useCategoriesCardData.tsx:22]: It was changed in [this PR](https://github.com/Jullija/oops/pull/40) so you might have to change that

File: frontend/src/screens/StudentProfile/StudentProfile.tsx [hunk 6]
```diff
@@ -12,16 +13,19 @@ const styles: Styles = {
...
+  const { user } = useUser();
+  const { categories, student, loading, error } = useStudentProfileData(
+    user.userId,
+  );
 
   // TODO: add components for loading state and error message
   if (loading) return <p>Loading...</p>;
   if (error) return <p>Error: {error.message}</p>;
-  if (!student) return <p>Please select an edition.</p>;
+  if (!student) return <p>student is undefined</p>;
```
Comment (Szyntos) [inline frontend/src/screens/StudentProfile/StudentProfile.tsx:24]: "Student" *

**Review by Szyntos — CHANGES_REQUESTED** [review]
Overall i don't see issues, good work!

You might need to change the not-null types tho

**Pull request merged**